tests/integration/fixtures/processing.py

```diff
@@ -167,6 +167,31 @@ def poll(self, timeout=None):
             timeout = self.timeout
         return self.consumer.poll(timeout=timeout)
 
+    def poll_many(self, timeout=None, n=None):
+        if timeout is None:
+            timeout = self.timeout
+
+        if n == 0:
+            self.assert_empty()
+            return
+
+        messages = 0
+
+        # Wait for the first outcome to show up for the full timeout duration
+        message = self.poll(timeout)
+        while message is not None:
+            yield message
+            messages += 1
+
+            if messages == n:
+                self.assert_empty()
+                break
+
+            message = self.poll(min(2, timeout) if n is None else timeout)
```

Comment:
Can't we use `min(2, timeout)` even if `n is not None`?

Reply:
It makes everything a bit more fragile then, if you expect a certain amount of items and they are not there we can afford to wait a little bit longer (should never happen). But if for some reason something actually is a bit slower we won't fail with the longer timeout.

Reply:
Makes sense.